CredentialProvider.Microsoft/CredentialProviders/Vsts/VstsCredentialProvider.cs

```diff
@@ -37,6 +37,20 @@ public sealed class VstsCredentialProvider : CredentialProviderBase
 
         public override async Task<bool> CanProvideCredentialsAsync(Uri uri)
         {
+            string buildTaskJsonEnvVar = Environment.GetEnvironmentVariable(EnvUtil.BuildTaskExternalEndpoints);
+            string buildTaskPrefixesEnvVar = Environment.GetEnvironmentVariable(EnvUtil.BuildTaskUriPrefixes);
+            string buildTaskAccessTokenEnvVar = Environment.GetEnvironmentVariable(EnvUtil.BuildTaskAccessToken);
+
+            bool shouldUseBuildTaskCredProvider = string.IsNullOrWhiteSpace(buildTaskJsonEnvVar) == false 
+                || string.IsNullOrWhiteSpace(buildTaskPrefixesEnvVar) == false 
+                || string.IsNullOrWhiteSpace(buildTaskAccessTokenEnvVar) == false;
+            if (shouldUseBuildTaskCredProvider)
```

Comment:
I thought the control first goes to build task cred provider. If it has already handled the request, why would the request reach this stage?

Reply:
There were cases where CanProvideCredentials check would return false from the build task cred providers even when the build task environment variables were set, and I don't think this cred provider should be tried even as a backup to avoid confusion in the possible error logs. As an alternative, I could only check the env vars in the build task CanProvideCredentials methods and move the other checks to the HandleRequestAsync method. I think I like that more - I'll make the change. This snippet here might just confuse people. Thanks!

Reply:
done in commit https://github.com/Microsoft/artifacts-credprovider/pull/50/commits/7d44855eb772c504aadb54d8eab8151389de4ec8